proto/v1beta1/run_function.proto

```diff
@@ -63,6 +63,19 @@ message RunFunctionRequest {
   // and that context will be passed to subsequent Functions. Crossplane
   // discards all context returned by the last Function in the pipeline.
   optional google.protobuf.Struct context = 5;
+
+  // Optional extra resources that the Function required.
+  // Note that extra resources is a map to Resources, plural.
+  // The map key corresponds to the key in a RunFunctionResponse's
+  // extra_resources field. If a Function requested extra resources that
+  // did not exist, Crossplane sets the map key to an empty Resources message to
+  // indicate that it attempted to satisfy the request.
+  map<string, Resources> extra_resources = 6;
+}
+
+// Resources just exists because you can't have map<string, repeated google.protobuf.Struct>.
+message Resources {
+  repeated google.protobuf.Struct items = 1;
```

Comment:
I wonder if there's any value using `repeated Resource` here. That would leave room for sending connection details for each extra resource (and potentially other fields added in future).

Reply:
that was my initial implementation too, and `Resources` not being `repeated Resource` triggers me a bit too, but I decided not to go that route because I couldn't think of a reason for connection details and readiness to be there. I could go either ways.

Reply:
I went with `Resources`, the inconsistency was more painful than the additional potentially useless fields.